public/docs/js/search.js

```diff
@@ -456,69 +541,103 @@ async function search(s, r) {
+        const segments = address.pathname.split('/').filter(Boolean);
+
+        segments.forEach((segment, index) => {
+            const words = segment.replace(/-/g, ' ').split(' ');
+            const processedSegment = words.map((word, index) => 
+                index === 0 ? word.charAt(0).toUpperCase() + word.slice(1).toLowerCase() : word.toLowerCase()
+            ).join(' ');
+        
+            const segmentSpan = document.createElement('span');
+            segmentSpan.className = 'result-path__segment';
+            segmentSpan.textContent = processedSegment;
+            path.appendChild(segmentSpan);
+        
+            if (index < segments.length - 1) {
+                const svgIcon = document.createElement('span');
+                svgIcon.className = 'result-path__icon';
+                svgIcon.innerHTML = `
+                    <svg xmlns="http://www.w3.org/2000/svg" width="6" height="10" viewBox="0 0 6 10" fill="none">
+                        <path d="M1 9L5 5L1 1" stroke="#7C98B4" stroke-width="1.6" stroke-linecap="round" stroke-linejoin="round"/>
+                    </svg>
+                `;
+                path.appendChild(svgIcon);
+            }
+        });
+
+        const listElementDescription = document.createElement('p');
+        listElementDescription.className = 'result-text';
         // Only highlight user query terms, not stemmed terms
-        markers.innerHTML = highlight(needle.description, queryTerms);
+        listElementDescription.innerHTML = highlight(needle.description, queryTerms);
 
         const li = document.createElement('li');
+        li.classList.add('site-search-results__item');
         li.dataset.words = needle.foundWords.toString();
         li.dataset.score = (Math.round((needle.score/ total) * 1000) / 1000).toString();
-        li.appendChild(a);
-        li.appendChild(path);
-        li.appendChild(markers);
-
-        if (needle.matchedHeadings.length > 0) {
-            const headings = document.createElement('ul');
-            headings.className = 'result-headings';
-
-            headings.tabIndex = 0;
-
-            needle.matchedHeadings
-                .forEach(h => {
-                    const item = document.createElement('li');
-                    const link = document.createElement('a');
-                    link.href = url + '#' + h.slug;
-                    // Only highlight user query terms, not stemmed terms
-                    link.innerHTML = highlight(h.text, queryTerms);
-                    item.appendChild(link);
-                    headings.append(item);
-                });
-
-            li.appendChild(headings);
-        }
-
-        ol.appendChild(li);
+        listElementWrapper.appendChild(path);
+        listElementWrapper.appendChild(listElementTitle);
+        listElementWrapper.appendChild(listElementDescription);
+        li.appendChild(listElementWrapper);
+
+        // TODO: Feature flag
```

Comment:
@steve-fenton-octopus 👋  Following your suggestion, I'm looking into using feature flags for the section I commented out. Could you provide some guidance on how to implement this 😅? Thanks!

Reply:
I've added a feature flag for whether to show headings in the search results. In docs, you can remove the headings by removing the 'headings' flag in src/config.ts:

```
	featureFlags: {
		stickyNav: { top: 112 },
		codeBlocks: ['copy'],
		figures: ['enlarge'],
		youTubeLinks: ['embed'],
		headers: ['link'],
		details: ['tabs'],
		search: ['dialog', 'headings'],
	},
```

So the search flags would become: `search: ['dialog'],`